### PR TITLE
Add a comment explaining why GC.AllocateUninitializedArray is not an icall

### PR DESCRIPTION
**Pull request by naricc**
I experimentally determined that implementing GC.AllocateUninitializedArray() is unlikely to save any time. To do this I constructed a program that allocates 100k arrays of the maximum size for which we do explicit zeroing, and modified the VM to skip zeroing. This resulted in no measurable speed up.

This change is a result of this issue: https://github.com/mono/mono/issues/15235

### CODE REVIEW

File: netcore/System.Private.CoreLib/src/System/GC.cs [hunk 1]
```diff
@@ -244,7 +244,11 @@ public static GCMemoryInfo GetGCMemoryInfo ()
 
 		internal static T[] AllocateUninitializedArray<T> (int length)
 		{
-			// TODO: Implement
+			// Mono only does explicit zeroning if the array is to big for the nursery, but less than 1 Mb - 4 kb.
+			// If it is bigger than that, we grab memroy directly from the OS which comes pre zeroed.
+			// Experimentation shows that if we just skip the zeroing in this case, we do not save a measurable
+			// amount of time. So we just allocate the normal way here.
+			// Revist if we change LOS implementation.
```
Comment (Therzok) [inline netcore/System.Private.CoreLib/src/System/GC.cs]:
```suggestion
			// Revisit if we change LOS implementation.
```

**Comment by naricc** [comment]
@monojenkins build failed

File: netcore/System.Private.CoreLib/src/System/GC.cs [hunk 2]
```diff
@@ -244,7 +244,11 @@ public static GCMemoryInfo GetGCMemoryInfo ()
 
 		internal static T[] AllocateUninitializedArray<T> (int length)
 		{
-			// TODO: Implement
+			// Mono only does explicit zeroning if the array is to big for the nursery, but less than 1 Mb - 4 kb.
+			// If it is bigger than that, we grab memroy directly from the OS which comes pre zeroed.
```
Comment (lambdageek) [inline netcore/System.Private.CoreLib/src/System/GC.cs]:
```suggestion
			// If it is bigger than that, we grab memory directly from the OS which comes pre zeroed.
```

File: netcore/System.Private.CoreLib/src/System/GC.cs [hunk 3]
```diff
@@ -244,7 +244,11 @@ public static GCMemoryInfo GetGCMemoryInfo ()
 
 		internal static T[] AllocateUninitializedArray<T> (int length)
 		{
-			// TODO: Implement
+			// Mono only does explicit zeroning if the array is to big for the nursery, but less than 1 Mb - 4 kb.
```
Comment (lambdageek) [inline netcore/System.Private.CoreLib/src/System/GC.cs]:
```suggestion
			// Mono only does explicit zeroning if the array is too big for the nursery, but less than 1 Mb - 4 kb.
```

Comment (filipnavara) [reply netcore/System.Private.CoreLib/src/System/GC.cs]: Also, s/zeroning/zeroing/.

**Pull request merged**